functions/Invoke-DbaQuery.ps1

```diff
@@ -267,6 +267,10 @@ function Invoke-DbaQuery {
             Stop-Function -Category InvalidArgument -Message "You can't use -SqlInstance with piped databases"
             return
         }
+        if (Test-Bound -ParameterName "SqlInstance" -Not) {
```

Comment:
Thanks so much for the PR, Richard! This is almost right. Because we allow piping in databases, SqlInstance _or_ InputObject is acceptable. Objects from the pipeline cannot be seen by begin, so this block will have to be moved down to `process` and it'll have to check for not sqlinstance and not inputobject. 

You may notice the tests failed due to this reason. Once the test passes you should be set 💯

Reply:
I think I have this correct. I am going to add the commit.